budgetApp1/budgetApp1/UserViewModel.swift

```diff
@@ -2,20 +2,23 @@ import Foundation
 
 class UserViewModel {
     
-    // MARK: - Singleton instantiation of ViewModel
-    static let shared = UserViewModel()
+    let authenticationRepository: AuthenticationRepositoryType
     
-    private var users: [User] = []
+    private var userData = [UserEntity]()
     
-    private init() {}
+    init(authenticationRepository: AuthenticationRepositoryType) {
+        self.authenticationRepository = authenticationRepository
+    }
+    
+    func fetchAllUsers() -> Result<[UserEntity], Error> {
+        authenticationRepository.fetchAllUsers()
```

Comment:
do you not want to set the result of users to the array in here?

Reply:
Done. That is a good suggestion!